packages/edit-site/src/components/app/index.js

```diff
@@ -48,12 +54,37 @@ export default function App() {
 			)
 		);
 	}
+	const middlewares = useMemo(
```

Comment:
`middleware` is a very generic name that doesn't tell you at all what it does. Here it is effectively a `beforeNavigate` filter. Can modify the args, but cannot abort the navigation (maybe it could, it's common for WP filters to do such things). Could it be exposed as `beforeNavigate` callback?

Reply:
Sure that works too.

Reply:
I didn't implement aborting navigation for now. Let's see if we need it first.